crates/biome_js_analyze/tests/specs/nursery/noImplicitCoercion/invalid.js

```diff
@@ -0,0 +1,31 @@
+!!foo;
+!!(foo + bar);
+!!(foo + bar);
+~foo.indexOf(1);
+~foo.bar.indexOf(2);
++foo;
+-(-foo);
++foo.bar;
+1*foo;
+foo*1;
+1*foo.bar;
+foo.bar-0;
+""+foo;
+    ``+foo;
+foo+"";
+foo+``;
+""+foo.bar;
+    ``+foo.bar;
+foo.bar+"";
+foo.bar+``;
+foo += "";
+foo += ``;
+typeof+foo;
+typeof +foo;
+let x ='' + 1n;
+~foo?.indexOf(1);
+~(foo?.indexOf)(1);
+1 * a / 2;
+(a * 1) / 2;
+a * 1 / (b * 1);
+/* test */ a * 1 + 2;
```

Comment:
If possible, we could add another test with a trailing comment 

```js
/* test */ a /*comment*/ * 1 + 2;
```

Reply:
Good case, added.